frontend/src/metabase/dashboard/containers/DashboardSharingEmbeddingModal/DashboardSharingEmbeddingModal.tsx

```diff
@@ -0,0 +1,67 @@
+import type { Dashboard } from "metabase-types/api";
+import type { EmbedOptions } from "metabase-types/store";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import * as Urls from "metabase/lib/urls";
+import { EmbedModal } from "metabase/public/components/widgets/EmbedModal";
+import EmbedModalContent from "metabase/public/components/widgets/EmbedModalContent";
+import { getParameters } from "metabase/dashboard/selectors";
+
+import {
+  createPublicLink,
+  deletePublicLink,
+  updateEnableEmbedding,
+  updateEmbeddingParams,
+} from "../../actions";
+
+type DashboardSharingEmbeddingModalProps = {
+  isLinkEnabled: boolean;
+  className?: string;
+  dashboard: Dashboard;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export const DashboardSharingEmbeddingModal = (
+  props: DashboardSharingEmbeddingModalProps,
+) => {
+  const { className, dashboard, isOpen, onClose, isLinkEnabled } = props;
+
+  const parameters = useSelector(state => getParameters(state));
+
+  const dispatch = useDispatch();
+
+  const createPublicDashboardLink = () => dispatch(createPublicLink(dashboard));
```

Comment:
We can most likely start removing these methods from the modal since they will be encapsulated into the different public link creation components, but for now I'll keep them here since there's a ton of prop spreading and a deeper investigation should be done